docs/topics/class-methods.rst

```diff
@@ -0,0 +1,29 @@
+===========================
+Class Instantiation Methods
+===========================
+
+These methods create an instance of their implementer class by 
+extracting the components needed for it from the argument that the method takes.
+
+.. py:classmethod:: from_crawler(cls, crawler)
+
+    Factory method that if present, is used to create an instance of the implementer class
```

Comment:
We need to be consistent with vocabulary. We should call them either ‘instantiation methods’ or ‘factory methods’, but we should stick to 1 name. I am slightly in favor of the later, but I have no strong opinion on it.